kythe/docs/schema/indexing-protobuf.txt

```diff
@@ -31,12 +31,235 @@ implementation, as this is often difficult (or uninteresting) to read. This
 document describes how to encode information about generated code to permit
 cross-language links.
 
+To make the discussion easier to understand let's pretend we are working with
+two languages: AliceLang and BobLang. AliceLang has `.alice` file and BobLang
+has `.bob` files. We also have a tool (generator) that can take generate
+`foo.bob` file from `foo.alice` file. We have following components:
+
+* Alice Indexer - Kythe indexer that takes `.alice` files and outputs index
+  data.
+* Bob Indexer - Kythe indexer that takes `.bob` files and outputs index
+  data.
+* Generator - tool that produces `.bob` files from `.alice` files.
+* Post processor - Kythe tool that takes all index data produced by all
+  indexers, processes it and outputs final Kythe graph that contains data
+  for both AliceLang and BobLang.
+
+Now we want to teach Kythe how to create cross-references between generated
+`foo.bob` file and original `foo.alice` file. The main idea is pretty simple:
+Generator has to output extra data containing mapping of elements in `foo.bob`
+to the original elements from `alice.bob`. Then when Bob Indexer is indexing
+`foo.bob` it will use that mapping to output *generates* or *imputes* edges.
+These edges connect nodes from `foo.bob` with nodes in `foo.alice`.
+
+Kythe doesn't require implementors to use one concrete approach for passing
+mapping metadata and outputting *generates* and *imputes* edges. Below we
```

Comment:
Not sure if this is mentioned elsewhere, but it might be worth mentioning why one might be preferred to the other.

Reply:
That's a good question. Which one is preferred though? :) Does Kythe team prefers `imputes` approach because it's more general and doesn't require coupling between indexers and generators?

Reply:
`generates` is slightly preferred because there's less work during post-processing, `imputes` exists to cover the case where the semantic VNames of the source language aren't stable/

Reply:
Added comment. Wondering if I should add a special section at the end of doc that compares both approaches and goes into depth of why both are needed.